sbin/wifibox

```diff
@@ -734,9 +740,39 @@ vm_manager() {
 
     log debug "Nice priority: ${_nice_priority}"
     log debug "Arguments: ${_bhyve_args}"
-    ${NICE} -n ${_nice_priority} \
-	    ${BHYVE} ${_bhyve_args} 2>&1 | capture_output debug bhyve
+
+    _kill_child() {
+    	local _job_pids
+    	_job_pids="$(jobs -p)"
+
+    	log debug "SIGTERM received for PID $PPID, killing remaining jobs"
+
+    	if [ -n "${_job_pids}" ]; then
+    	    for job_pid in ${_job_pids}; do
+    		log debug "job_pid=${job_pid}"
+    		${KILL} -SIGTERM ${job_pid} | capture_output debug kill
+    	    done
+    	fi
+    }
+
+    trap _kill_child SIGTERM
+
+    { ${NICE} -n ${_nice_priority} \
+	    ${BHYVE} ${_bhyve_args} 2>&1 | capture_output debug bhyve; } &
+    _nice_pid=$!
+
+    log debug "Waiting for VM to stop"
+
+    wait "$_nice_pid"
     _bhyve_exit_code="$?"
+
+    trap - SIGTERM
+
+    if [ "${_bhyve_exit_code}" -eq "143" ]; then
```

Comment:
Why do we need this condition here?

Reply:
@MegaManSec could you please answer this question?

Reply:
Looks like it got stuck in a review instead of a comment:
![Screenshot-2024-09-05:09:22:53](https://github.com/user-attachments/assets/51fa4282-5a71-49c7-b90c-cf6a0d761481)

Reply:
Minor note: we are not working with `bash` here.  The `wifibox` script is using FreeBSD's default POSIX-compatible system shell (+ Berkeley extensions), which is `sh`.

Reply:
Otherwise, I am fine with this solution.

Reply:
yes sorry, i just typed it in my normal terminal and thought the error message was helpful for demonstrative purposes